### PR TITLE
Cleanup to various loc assignments and nearby code

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
I tested some of the more extensive changes, particularly buildandrepair.dm, and everything *seems* to work...

[why]: 
Code improvement

### CODE REVIEW

File: code/datums/antagonists/devil.dm [hunk 1]
```diff
@@ -205,7 +205,7 @@ GLOBAL_LIST_INIT(devil_suffix, list(" the Red", " the Soulless", " the Master",
 /datum/antagonist/devil/proc/regress_blood_lizard()
 	var/mob/living/carbon/true_devil/D = owner.current
 	to_chat(D, "<span class='warning'>Your powers weaken, have more contracts be signed to regain power.</span>")
-	D.oldform.loc = D.loc
+	D.oldform.forceMove(D.loc)
```
Comment (Cyberboss) [inline code/datums/antagonists/devil.dm]: D.drop_location()

File: code/datums/diseases/parrotpossession.dm [hunk 2]
```diff
@@ -22,6 +22,6 @@
 
 /datum/disease/parrot_possession/cure()
 	if(parrot && parrot.loc == affected_mob)
-		parrot.loc = affected_mob.loc
+		parrot.forceMove(affected_mob.loc)
```
Comment (Cyberboss) [inline code/datums/diseases/parrotpossession.dm]: affected_mob.drop_location()

File: code/datums/wires/wires.dm [hunk 3]
```diff
@@ -153,7 +153,7 @@
 	if(S && istype(S))
 		assemblies -= color
 		S.connected = null
-		S.loc = holder.loc
+		S.forceMove(holder.loc)
```
Comment (Cyberboss) [inline code/datums/wires/wires.dm]: holder.drop_location()

File: code/game/gamemodes/blob/powers.dm [hunk 4]
```diff
@@ -59,7 +59,7 @@
 	set name = "Jump to Core"
 	set desc = "Move your camera to your core."
 	if(blob_core)
-		src.loc = blob_core.loc
+		src.forceMove(blob_core.loc)
```
Comment (Cyberboss) [inline code/game/gamemodes/blob/powers.dm]: remove src

Comment (Cyberboss) [inline code/game/gamemodes/blob/powers.dm]: blob_core.drop_location()

File: code/game/gamemodes/miniantags/abduction/machinery/dispenser.dm [hunk 5]
```diff
@@ -62,7 +62,7 @@
 	if(istype(W, /obj/item/organ/heart/gland))
 		if(!user.drop_item())
 			return
-		W.loc = src
+		W.forceMove(src)
```
Comment (Cyberboss) [inline code/game/gamemodes/miniantags/abduction/machinery/dispenser.dm]: if(!user.transferItemToLoc(src))
    return

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 6]
```diff
@@ -502,7 +502,7 @@
 		var/obj/item/stack/sheet/metal/M = new /obj/item/stack/sheet/metal(target.loc)
 		M.amount = 5
 		for(var/obj/item/I in target.component_parts)
-			I.loc = M.loc
+			I.forceMove(M.loc)
```
Comment (Cyberboss) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm]: M.drop_location()

File: code/game/gamemodes/miniantags/bot_swarm/swarmer.dm [hunk 7]
```diff
@@ -511,7 +511,7 @@
 		if(istype(target, /obj/machinery/computer))
 			var/obj/machinery/computer/C = target
 			if(C.circuit)
-				C.circuit.loc = M.loc
+				C.circuit.forceMove(M.loc)
```
Comment (Cyberboss) [inline code/game/gamemodes/miniantags/bot_swarm/swarmer.dm]: M.drop_location()

File: code/game/gamemodes/nuclear/nuclear.dm [hunk 8]
```diff
@@ -132,7 +132,7 @@
 		P.info = "The nuclear authorization code is: <b>[nuke_code]</b>"
 		P.name = "nuclear bomb code"
 		var/mob/living/carbon/human/H = synd_mind.current
-		P.loc = H.loc
+		P.forceMove(H.loc)
```
Comment (Cyberboss) [inline code/game/gamemodes/nuclear/nuclear.dm]: H.drop_location()

File: code/game/gamemodes/sandbox/h_sandbox.dm [hunk 9]
```diff
@@ -140,7 +140,7 @@ GLOBAL_VAR_INIT(hsboxspawn, TRUE)
 				var/mob/living/carbon/human/P = usr
 				if(!istype(P)) return
 				if(P.wear_suit)
-					P.wear_suit.loc = P.loc
+					P.wear_suit.forceMove(P.loc)
```
Comment (Cyberboss) [inline code/game/gamemodes/sandbox/h_sandbox.dm]: P.drop_location()

File: code/game/gamemodes/sandbox/h_sandbox.dm [hunk 10]
```diff
@@ -149,7 +149,7 @@ GLOBAL_VAR_INIT(hsboxspawn, TRUE)
 				P.wear_suit.plane = ABOVE_HUD_PLANE
 				P.update_inv_wear_suit()
 				if(P.head)
-					P.head.loc = P.loc
+					P.head.forceMove(P.loc)
```
Comment (Cyberboss) [inline code/game/gamemodes/sandbox/h_sandbox.dm]: P.drop_location()

File: code/game/gamemodes/sandbox/h_sandbox.dm [hunk 11]
```diff
@@ -158,7 +158,7 @@ GLOBAL_VAR_INIT(hsboxspawn, TRUE)
 				P.head.plane = ABOVE_HUD_PLANE
 				P.update_inv_head()
 				if(P.wear_mask)
-					P.wear_mask.loc = P.loc
+					P.wear_mask.forceMove(P.loc)
```
Comment (Cyberboss) [inline code/game/gamemodes/sandbox/h_sandbox.dm]: P.drop_location()

File: code/game/gamemodes/sandbox/h_sandbox.dm [hunk 12]
```diff
@@ -167,7 +167,7 @@ GLOBAL_VAR_INIT(hsboxspawn, TRUE)
 				P.wear_mask.plane = ABOVE_HUD_PLANE
 				P.update_inv_wear_mask()
 				if(P.back)
-					P.back.loc = P.loc
+					P.back.forceMove(P.loc)
```
Comment (Cyberboss) [inline code/game/gamemodes/sandbox/h_sandbox.dm]: P.drop_location()

File: code/game/machinery/camera/camera.dm [hunk 13]
```diff
@@ -246,7 +246,7 @@
 		if(disassembled)
 			if(!assembly)
 				assembly = new()
-			assembly.loc = src.loc
+			assembly.forceMove(src.loc)
```
Comment (Cyberboss) [inline code/game/machinery/camera/camera.dm]: drop_location()

Comment (Cyberboss) [inline code/game/machinery/camera/camera.dm]: Remove src

File: code/game/machinery/camera/camera_assembly.dm [hunk 14]
```diff
@@ -91,7 +91,7 @@
 
 				state = 4
 				var/obj/machinery/camera/C = new(src.loc)
-				src.loc = C
+				src.forceMove(C)
```
Comment (Cyberboss) [inline code/game/machinery/camera/camera_assembly.dm]: remove src

File: code/game/machinery/camera/camera_assembly.dm [hunk 15]
```diff
@@ -122,7 +122,7 @@
 		if(U)
 			to_chat(user, "<span class='notice'>You unattach an upgrade from the assembly.</span>")
 			playsound(src.loc, W.usesound, 50, 1)
-			U.loc = get_turf(src)
+			U.forceMove(get_turf(src))
```
Comment (Cyberboss) [inline code/game/machinery/camera/camera_assembly.dm]: drop_location()

File: code/game/machinery/computer/buildandrepair.dm [hunk 16]
```diff
@@ -40,31 +40,30 @@
 					state = 0
 				return
 			if(istype(P, /obj/item/circuitboard/computer) && !circuit)
-				if(!user.drop_item())
+				if(!user.transferItemToLoc(P, null))
```
Comment (Cyberboss) [inline code/game/machinery/computer/buildandrepair.dm]: use temporarilyRemoveItemFromInventory to put stuff in nullspace

File: code/game/machinery/computer/card.dm [hunk 17]
```diff
@@ -54,22 +54,19 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 		var/obj/item/card/id/idcard = O
 		if(check_access(idcard))
 			if(!scan)
-				if(!usr.drop_item())
+				if (!usr.transferItemToLoc(idcard,src))
```
Comment (Cyberboss) [inline code/game/machinery/computer/card.dm]: user

File: code/game/machinery/computer/card.dm [hunk 18]
```diff
@@ -54,22 +54,19 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
...
 			if(!scan)
-				if(!usr.drop_item())
+				if (!usr.transferItemToLoc(idcard,src))
 					return
-				idcard.loc = src
 				scan = idcard
 				playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
 			else if(!modify)
-				if(!usr.drop_item())
+				if (!usr.transferItemToLoc(idcard,src))
```
Comment (Cyberboss) [inline code/game/machinery/computer/card.dm]: user

File: code/game/machinery/computer/card.dm [hunk 19]
```diff
@@ -54,22 +54,19 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
...
 					return
-				idcard.loc = src
 				modify = idcard
 				playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
 		else
 			if(!modify)
-				if(!usr.drop_item())
+				if (!usr.transferItemToLoc(idcard,src))
```
Comment (Cyberboss) [inline code/game/machinery/computer/card.dm]: user

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/card.dm [hunk 20]
```diff
@@ -355,7 +352,7 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 			if (modify)
 				GLOB.data_core.manifest_modify(modify.registered_name, modify.assignment)
 				modify.update_label()
-				modify.loc = loc
+				modify.forceMove(loc)
```
Comment (Cyberboss) [inline code/game/machinery/computer/card.dm]: drop_location()

File: code/game/machinery/computer/card.dm [hunk 21]
```diff
@@ -364,26 +361,24 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
...
 						return
 					playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
-					I.loc = src
 					modify = I
 			authenticated = 0
 
 		if ("scan")
 			if (scan)
-				scan.loc = src.loc
+				scan.forceMove(src.loc)
```
Comment (Cyberboss) [inline code/game/machinery/computer/card.dm]: drop_location()

Comment (Cyberboss) [reply code/game/machinery/computer/card.dm]: remove src

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by vuonojenmustaturska** [comment]
I'll get on it tomorrow

**Comment by optimumtact** [comment]
apparently I was tagged in this?

**Comment by vuonojenmustaturska** [comment]
Yeah there was a review about waiting for your movetonullspace thing

**Review by Cyberboss — DISMISSED** [review]

File: code/game/machinery/_machinery.dm [hunk 22]
```diff
@@ -429,7 +429,7 @@ Class Procs:
 							W.handle_item_insertion(A, 1)
 							component_parts -= A
 							component_parts += B
-							B.loc = null
+							B.forceMove(null)
```
Comment (Cyberboss) [inline code/game/machinery/_machinery.dm]: Needs #31071

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**